etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,516 @@
+	// +optional
+	ActiveStatus ResourceActiveStatusType `json:"activeStatus,omitempty"`
+
+	// node is the node where the resource is running
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the resource is not assigned to a node. This typically indicates a stopped or unscheduled resource. It could also imply an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Node string `json:"node,omitempty"`
+}
+
+// PacemakerNodeHistoryEntry represents a single operation history entry from node_history
+type PacemakerNodeHistoryEntry struct {
+	// node is the node where the operation occurred
+	// It must be a valid string between 1 and 256 characters long and cannot be empty.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Node string `json:"node,omitempty"`
+
+	// resource is the resource that was operated on
+	// It must be a valid string between 1 and 256 characters long and cannot be empty.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Resource string `json:"resource,omitempty"`
+
+	// operation is the operation that was performed (e.g., "monitor", "start", "stop")
+	// Unlike other fields, this is not an enum because while "monitor", "start" and "stop"
+	// are the most common, resource agents can define their own operations.
+	// It must be a valid string between 1 and 32 characters long and cannot be empty.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=32
+	// +required
+	Operation string `json:"operation,omitempty"`
+
+	// rc is the return code from the operation
+	// When present, it must be a valid integer between 0 and 2147483647 (max 32-bit int) inclusive.
+	// When not present, the return code is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// +optional
+	RC *int32 `json:"rc,omitempty"`
+
+	// rcText is the human-readable return code text (e.g., "ok", "error", "not running")
+	// When present, it must be a valid string between 1 and 32 characters long. This is a human-readable string and is not validated against any specific format.
+	// When not present, the return code text is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=32
+	// +optional
+	RCText string `json:"rcText,omitempty"`
+
+	// lastRCChange is the timestamp when the RC last changed
+	// It must be a valid timestamp in RFC3339 format and cannot be empty.
+	// +kubebuilder:validation:Format=RFC3339
+	// +required
+	LastRCChange metav1.Time `json:"lastRCChange,omitempty"`
+}
```

Comment:
This feels like it would be better represented as an emitted corev1.Event

Reply:
The new working model is to avoid storing things in the API that just exist to create other things. E.g. events